src/bsp/src/stm32/src/SocketContainer.cpp

```diff
@@ -11,13 +12,11 @@ std::optional<TCPClientWrapper> SocketContainer::getHostClientSocket() {
     if (!s_clientSocket) {
 
         ILogger& logger = LoggerContainer::getLogger();
+        std::string hostAddress = SettingsContainer::GetHostIP();
```

Comment:
Don't forget to change here for a const char* after the change on GetHostIp so you don't use a std::string constructor